src/luxonis_ml/utils/logging.py

```diff
@@ -0,0 +1,75 @@
+import logging
+import os
+import warnings
+from typing import Literal
+
+from rich.console import Console
+from rich.logging import RichHandler
+from rich.theme import Theme
+
+
+def setup_logging(
+    *,
+    file: str | None = None,
+    use_rich: bool = False,
+    level: Literal["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"] = "INFO",
+    configure_warnings: bool = True,
+) -> None:
+    """Globally configures logging.
+
+    Configures a standar Luxonis logger.
+    Optionally utilizes rich library, configures handling of warnings
+    (from warnings module) and saves the logs to a file.
+
+    Args:
+        file (str, optional): Path to a file where logs will be saved.
+          If None, logs will not be saved. Defaults to None.
+        use_rich (bool, optional): If True, rich library will be used for logging.
+          Defaults to False.
+        level (str, optional): Logging level. One of "DEBUG", "INFO", "WARNING",
+          "ERROR", and "CRITICAL". Defaults to "INFO".
+          If "DEBUG" is set in environment variables, the level is changed to "DEBUG".
+        configure_warnings (bool, optional): If True, warnings will be logged.
+          Defaults to True.
+    """
+    # NOTE: So we can simply run `DEBUG= python ...`
+    if "DEBUG" in os.environ:
```

Comment:
Would it make sense having some dedicated variable that can set either of the levels instead?